### PR TITLE
Add option to use cloud and integrate signup/login with local DB

### PR DESCRIPTION
**Pull request by Carson-Tang**
this is such bad code

in order to see local database see images below

image
image

### CODE REVIEW

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/SignUpCustomizationPages.kt [hunk 1]
```diff
@@ -115,19 +115,33 @@ fun SignUpPIN(pageState: MutableState<PageStates>) {
...
+        else ""
     ElevatedButton(
-        onClick = { pageState.value = nextPageState },
+        onClick = {
+            pageState.value = nextPageState
+            if (pageState.value == PageStates.SIGNUP_CLOUD) {
+                useCloud.value = true
+            }
+            else if (pageState.value == PageStates.SIGNUP_AFFIRMATION) {
+                useJournalForAffirmations.value = false
```
Comment (Carson-Tang) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/SignUpCustomizationPages.kt:142]:
```suggestion
                useJournalForAffirmations.value = true
```

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/SignUpCustomizationPages.kt [hunk 2]
```diff
@@ -115,19 +115,33 @@ fun SignUpPIN(pageState: MutableState<PageStates>) {
...
     ElevatedButton(
-        onClick = { pageState.value = nextPageState },
+        onClick = {
+            pageState.value = nextPageState
+            if (pageState.value == PageStates.SIGNUP_CLOUD) {
+                useCloud.value = true
+            }
+            else if (pageState.value == PageStates.SIGNUP_AFFIRMATION) {
+                useJournalForAffirmations.value = true
+            }
```
Comment (Carson-Tang) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/SignUpCustomizationPages.kt:143]:
```suggestion
            if (pageState.value == PageStates.SIGNUP_CLOUD) {
                useCloud.value = true
            }
            else if (pageState.value == PageStates.SIGNUP_AFFIRMATION) {
                useJournalForAffirmations.value = true
            }
            pageState.value = nextPageState
```

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/SignUpCustomizationPages.kt [hunk 3]
```diff
@@ -141,9 +155,19 @@ fun CustomizationActionButtons(
...
     if (pageState.value != PageStates.SIGNUP_CLOUD_MORE) {
-        TextButton(onClick = { pageState.value = nextPageState }) {
+        TextButton(onClick = {
+            pageState.value = nextPageState
+            if (pageState.value == PageStates.SIGNUP_CLOUD) {
+                useCloud.value = false
+            }
+            else if (pageState.value == PageStates.SIGNUP_AFFIRMATION) {
+                useJournalForAffirmations.value = false
+            }
```
Comment (Carson-Tang) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/SignUpCustomizationPages.kt:165]:
```suggestion
            if (pageState.value == PageStates.SIGNUP_CLOUD) {
                useCloud.value = false
            }
            else if (pageState.value == PageStates.SIGNUP_AFFIRMATION) {
                useJournalForAffirmations.value = false
            }
            pageState.value = nextPageState
```

**Review by alice41abc — APPROVED** [review]

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/AuthPages.kt [hunk 4]
```diff
@@ -171,45 +187,93 @@ fun SignUpLoginPage(
...
-                                } else if (response.status == HttpStatusCode.OK) {
-                                    // TODO: something with jwt
-                                    pageState.value = PageStates.HOME
+                            // LOCAL
+                            var userRes: User? = null
+                            val job = GlobalScope.launch {
+                                userRes = userDao.findByEmail(emailState.value.trimEnd())
+                            }
+                            job.join()
+                            println("userRes" + (userRes?.password))
```
Comment (alice41abc) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/AuthPages.kt:253]: Should probably delete this

**Review by alice41abc — APPROVED** [review]

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/AuthPages.kt (hunk 4 above)
Comment (Carson-Tang) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/AuthPages.kt:253]:
```suggestion

```

**Pull request merged**